src/Beberlei/Metrics/Collector/Graphite.php

```diff
@@ -85,21 +89,18 @@ public function flush()
             return;
         }
 
-        try {
-            $fp = fsockopen($this->protocol.'://'.$this->host, $this->port);
-
-            if (!$fp) {
-                return;
-            }
+        $fp = @fsockopen($this->protocol . '://' . $this->host, $this->port, $errno, $errmsg, $this->timeout);
```

Comment:
using `@` is a bad idea. Could you use a custom error handler?
see this example: https://github.com/redirectionio/proxy-sdk-php/blob/master/src/Client.php#L231-L251

thanks